ui/store/actions.js

```diff
@@ -1105,6 +1112,28 @@ export function updateMetamaskState(newState) {
       type: actionConstants.UPDATE_METAMASK_STATE,
       value: newState,
     });
+
+    // Check that the transaction was not submitted successfully, and remove it from failed transactions if it was.
+
```

Comment:
Nit: This wording is a bit awkward (my bad, I think this is from an earlier suggestion of mine), and this newline isn't needed. Here is a suggested improvement.

 ```suggestion
    // Remove successful transactions from `transactionsToDisplayOnFailure`.
```